src/Traits/OAuth2/ClientCredentialsGrant.php

```diff
@@ -27,6 +27,8 @@ trait ClientCredentialsGrant
      *
      * @param array<string> $scopes
      * @param callable(TRequest): (void)|null $requestModifier
+     *
+     * @return ($returnResponse is true ? \Saloon\Http\Response : \Saloon\Contracts\OAuthAuthenticator)
```

Comment:
No need for Fully Qualified Domain Name as it's already used bellow in the signature:

```suggestion
     * @return ($returnResponse is true ? Response : OAuthAuthenticator)
```